rest-catalog/quarkus-rest-catalog/src/intTest/java/org/projectnessie/restcatalog/server/resources/NessieDockerTestResourceLifecycleManager.java

```diff
@@ -0,0 +1,241 @@
+    return nessie;
+  }
+
+  private DevServicesContext context;
+  private String dockerImage;
+  private String dockerTag;
+  private String dockerNetworkId;
+  private boolean authEnabled;
+  private boolean debugEnabled;
+  private Map<String, String> extraEnvVars;
+
+  @Override
+  public void setIntegrationTestContext(DevServicesContext context) {
+    this.context = context;
+  }
+
+  @Override
+  public void init(Map<String, String> initArgs) {
+    dockerImage = initArgs.get(NESSIE_DOCKER_IMAGE);
+    if (dockerImage == null) {
+      dockerImage = System.getProperty(NESSIE_DOCKER_IMAGE, "ghcr.io/projectnessie/nessie");
+    }
+    dockerTag = initArgs.get(NESSIE_DOCKER_TAG);
+    if (dockerTag == null) {
+      dockerTag = System.getProperty(NESSIE_DOCKER_TAG, "latest");
+    }
+    dockerNetworkId = initArgs.get(NESSIE_DOCKER_NETWORK_ID);
+    if (dockerNetworkId == null) {
+      dockerNetworkId = System.getProperty(NESSIE_DOCKER_NETWORK_ID);
+    }
+    if (initArgs.containsKey(NESSIE_DOCKER_AUTH_ENABLED)) {
+      authEnabled = Boolean.parseBoolean(initArgs.get(NESSIE_DOCKER_AUTH_ENABLED));
+    } else {
+      authEnabled = Boolean.getBoolean(NESSIE_DOCKER_AUTH_ENABLED);
+    }
+    if (initArgs.containsKey(NESSIE_DOCKER_DEBUG_ENABLED)) {
+      debugEnabled = Boolean.parseBoolean(initArgs.get(NESSIE_DOCKER_DEBUG_ENABLED));
+    } else {
+      debugEnabled = Boolean.getBoolean(NESSIE_DOCKER_DEBUG_ENABLED);
+    }
+    this.extraEnvVars =
+        initArgs.entrySet().stream()
+            .filter(e -> !e.getKey().startsWith("nessie.docker."))
+            .map(e -> Map.entry(asEnvVar(e.getKey()), e.getValue()))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+  }
+
+  private static String asEnvVar(String key) {
+    return key.replaceAll("[.\\-\"]", "_").toUpperCase();
+  }
+
+  @Override
+  public Map<String, String> start() {
+
+    LOGGER.info("Using Nessie image {}:{}", dockerImage, dockerTag);
+    nessie = new NessieContainer();
+
+    LOGGER.info("Starting Nessie container...");
+    nessie.start();
+
+    URI nessieUri = nessie.getExternalNessieUri();
+    LOGGER.info("Nessie container started, URI for external clients: {}", nessieUri);
+
+    return Map.of(
+        "nessie.iceberg.nessie-client.\"" + NessieConfigConstants.CONF_NESSIE_URI + "\"",
+        nessieUri.toString());
+  }
+
+  @Override
+  public void inject(TestInjector testInjector) {
+    testInjector.injectIntoFields(
+        nessie.getExternalNessieUri(),
+        new TestInjector.AnnotatedAndMatchesType(NessieDockerUri.class, URI.class));
+  }
+
+  @Override
+  public void stop() {
+    try {
+      NessieContainer nessie = NessieDockerTestResourceLifecycleManager.nessie;
+      if (nessie != null) {
+        nessie.stop();
+      }
+    } finally {
+      NessieDockerTestResourceLifecycleManager.nessie = null;
+    }
+  }
+
+  public class NessieContainer extends GenericContainer<NessieContainer> {
```

Comment:
Should this provide a `StartupCheckStrategy`?
E.g. trying a dummy HTTP request against the `/` URI path (aka index.html).

Reply:
We already have a WaitStrategy:

```java
waitingFor(Wait.forLogMessage(".*Nessie.*started.*Listening on.*", 1));
```

In which cases would this not be sufficient? I could see one case: if the OIDC backend is unreachable, then the Nessie container would start normally, but tests would fail. But OTOH it's non trivial to create a `StartupCheckStrategy` for that, we would need to ping a protected endpoint like `/api/v2/config` and check that we get a 200 back. Do you think it's worth the effort?

Reply:
Sorry - missed that one.